### PR TITLE
Convert Datastore Dashboard Controller into Form

### PR DESCRIPTION
**Pull request by clayliddell**
- [x] Test coverage exists
- [ ] Documentation exists

## QA Steps

- [ ] Create a dataset via a harvest file.
- [ ] Ensure the DKAN instance you are working with has at least 11 datasets.
- [ ] Navigate to the datastore dashboard.
- [ ] Ensure only 10 datasets are showing up on the first page, and other datasets can be accessed via the pager.
- [ ] Ensure the UUID filter is working on the datastore dashboard.
- [ ] Ensure the harvest filter is working on the datastore dashboard.
- [ ] Ensure links for harvest IDs on the harvest dashboard navigate to a pre-filtered datastore dashboard.

### CODE REVIEW

File: modules/datastore/src/Form/DashboardForm.php [hunk 1]
```diff
@@ -0,0 +1,440 @@
...
+      $rows = array_merge($rows, $datasetRow);
+    }
+
+    return $rows;
+  }
+
+  /**
+   * Fetch the status of all harvests.
+   *
+   * @return \Generator
```
Comment (dafeder) [inline modules/datastore/src/Form/DashboardForm.php]: @clayliddell idk if upgrading codeclimate will fix this, but really this is not a very useful piece of documentation anyway - it's technically correct but doesn't really tell you what is being yielded. I'd say just remove this line.

**Review by dafeder — CHANGES_REQUESTED** [review]
Let's override codeclimate on the buildFilters() method -- sometimes Drupal's just gonna Drupal -- but remove that `@return`

File: modules/datastore/src/Form/DashboardForm.php [hunk 2]
```diff
@@ -0,0 +1,440 @@
...
+   *   Dataset harvest statuses.
```
Comment (clayliddell) [inline modules/datastore/src/Form/DashboardForm.php]:
```suggestion
```

File: modules/datastore/src/Form/DashboardForm.php (hunk 1 above)
Comment (clayliddell) [reply modules/datastore/src/Form/DashboardForm.php]: Fair enough. Resolved in https://github.com/GetDKAN/dkan/pull/3714#discussion_r760346155.

File: modules/datastore/src/Form/DashboardForm.php [hunk 3]
```diff
@@ -0,0 +1,437 @@
...
+
+  /**
+   * Build datasets import status table filters.
+   *
+   * @param string[] $filters
+   *   Dataset filters.
+   *
+   * @return array[]
+   *   Table filters render array.
+   */
```
Comment (clayliddell) [inline modules/datastore/src/Form/DashboardForm.php:180]:
```suggestion
   *
   * @SuppressWarnings(PHPMD.ExcessiveMethodLength)
   */
```

Comment (dafeder) [reply modules/datastore/src/Form/DashboardForm.php:180]: hmm neat trick. ugly but effective!

Comment (dafeder) [reply modules/datastore/src/Form/DashboardForm.php:180]: heh nevermind, does not look like CC catches that

Comment (clayliddell) [reply modules/datastore/src/Form/DashboardForm.php:180]: Yeah, I've tried everything I can think of. I thought this particular warning was coming from PHPMD, but I guess not...

> Let's override codeclimate on the buildFilters() method -- sometimes Drupal's just gonna Drupal...

How I can do this?

**Pull request merged**